mmocr/models/common/backbones/clip_resnet.py

```diff
@@ -0,0 +1,70 @@
+# Copyright (c) OpenMMLab. All rights reserved.
+from typing import Tuple, Union
+
+import torch
+import torch.nn as nn
+from mmdet.models.backbones import ResNet
+from mmdet.models.backbones.resnet import Bottleneck
+
+from mmocr.registry import MODELS
+
+
+@MODELS.register_module()
+class AvgPool2d(nn.Module):
+
+    def __init__(self,
+                 kernel_size: Union[int, Tuple[int]],
+                 stride: Union[int, Tuple[int]] = None,
+                 padding: Union[int, Tuple[int]] = 0,
+                 **kwargs) -> None:
+        super().__init__()
+        self.model = nn.AvgPool2d(kernel_size, stride, padding)
+
+    def forward(self, x) -> torch.Tensor:
+        return self.model(x)
+
+
+class CLIPBottleneck(Bottleneck):
+
+    def __init__(self, **kwargs):
+        stride = kwargs.get('stride', 1)
+        kwargs['stride'] = 1
+        plugins = kwargs.get('plugins', None)
+        if stride > 1:
+            if plugins is None:
+                plugins = []
+
+            plugins.insert(
+                0,
+                dict(
+                    cfg=dict(type='AvgPool2d', kernel_size=2),
+                    position='after_conv2'))
+            kwargs['plugins'] = plugins
+        super().__init__(**kwargs)
+
+
+@MODELS.register_module()
+class CLIPResNet(ResNet):
+    r"""CLIPResNet is a variant ResNet used in `oCLIP
+    <https://github.com/bytedance/oclip>`_.
+
+    It is the official structure in `CLIP
+    <https://github.com/openai/CLIP>`_.
+    """
+    arch_settings = {
+        50: (CLIPBottleneck, (3, 4, 6, 3)),
+    }
+
+    def __init__(self, **kwargs):
+        super(CLIPResNet, self).__init__(
+            depth=50,
+            strides=(1, 2, 2, 2),
+            deep_stem=True,
+            avg_down=True,
+            **kwargs)
```

Comment:
When parameters such as `depth`, and `strides` are passed by config, it will cause an error. 
```suggestion
    def __init__(self, 
                 depth=50,
                 strides=(1, 2, 2, 2),
                 deep_stem=True，
                 avg_down=True,
                 **kwargs):
        super().__init__(
            depth=depth,
            strides=strides,
            deep_stem=deep_stem,
            avg_down=avg_down,
            **kwargs)
```

Reply:
done